tests/L0/run_transformer/test_pipeline_parallel_fwd_bwd.py

```diff
@@ -31,33 +31,39 @@
 
 weight_coeff = 1024
 
-@torch.no_grad()
-def init_weights(m):
-    rank = torch.distributed.get_rank()
-    if isinstance(m, torch.nn.Linear):
-        m.weight.fill_((rank + 1.0) / weight_coeff)
-        m.bias.fill_(1.0)
+def get_init_weights_func(offset: int = 0):
+    @torch.no_grad()
+    def init_weights(m):
+        rank = parallel_state.get_pipeline_model_parallel_rank()
+        if isinstance(m, torch.nn.Linear):
+            m.weight.fill_((rank + offset + 1.0) / weight_coeff)
+            m.bias.fill_(1.0)
+    return init_weights
 
 
-def get_target_loss(hidden_size: int, microbatch_size: int, parallel_model_world_size: int, world_size: int) -> float:  
-    layers_per_rank = world_size // parallel_model_world_size
-    data = torch.arange(start = 0, end = layers_per_rank, dtype = torch.int) + 1
+def get_target_loss(global_batch_shape: tuple, hidden_size: int, total_layers: int) -> float:  
+    model = []
+    data = torch.ones(global_batch_shape, dtype=torch.float)
+    for i in range(total_layers):
+        w = torch.ones((hidden_size, hidden_size)) * (i + 1.0) / weight_coeff
+        b = torch.ones(hidden_size)
 
-    w = (torch.arange(world_size, dtype = torch.float) + 1.0) / weight_coeff
-    b = torch.ones(world_size, dtype = torch.int)
-    w = hidden_size * w
+        w.requires_grad_()
+        b.requires_grad_()
 
-    for s_id in range(0, world_size, layers_per_rank):
-        e_id = s_id+layers_per_rank
-        data = w[s_id:e_id] * data + b[s_id:e_id]
+        # don't need to care about transpose semantics as all values are the same
+        data = torch.matmul(w, data) + b
+        model.append([w, b])
 
-    return hidden_size * hidden_size * torch.sum(data).item() * microbatch_size / layers_per_rank
+    loss = data.sum() / global_batch_shape[0]
+    loss.backward()
 
+    return float((data.sum()/global_batch_shape[0]).item()), model
 
 class PipelineParallelForwardBackwardTest(DistributedTestBase):
 
-    GLOBAL_BATCH_SIZE = 16
-    MICRO_BATCH_SIZE = 2
+    GLOBAL_BATCH_SIZE = 32
+    MICRO_BATCH_SIZE = 1
```

Comment:
[question] Could you tell me why these changes are needed?
(I admit that these values are magic number though)

Reply:
Sorry, I played with these numbers to make sure, that everything works fine and I forgot to change them to the original values. I reverted these changes in [d8e7171](https://github.com/NVIDIA/apex/pull/1386/commits/d8e71716026bf61daf5484653498228e7b18ddb4).